File: archives/C#/0705. Design HashSet.cs
public class MyHashSet {

    Dictionary<int,int> mySet=new Dictionary<int,int>();
    /** Initialize your data structure here. */
    public MyHashSet() {
        
    }
    
    public void Add(int key) {
        if(!mySet.ContainsKey(key)){
            mySet[key]=1;
        }
    }
    
    public void Remove(int key) {
        if(mySet.ContainsKey(key)){
            mySet[key]-=1;
            if (mySet[key]==0){
                mySet.Remove(key);
            }
        }
        
    }
    
    /** Returns true if this set contains the specified element */
    public bool Contains(int key) {
        return mySet.ContainsKey(key);
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.Add(key);
 * obj.Remove(key);
 * bool param_3 = obj.Contains(key);
 */

File: archives/20190519python/0705. Design HashSet.py
class MyHashSet:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.set=[0]*1000001

    def add(self, key: int) -> None:
        self.set[key]=1
        

    def remove(self, key: int) -> None:
        self.set[key]=0

    def contains(self, key: int) -> bool:
        """
        Returns true if this set contains the specified element
        """
        if self.set[key]==1:
            return True
        return False
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)
